prober/tls.go

```diff
@@ -106,50 +104,49 @@ func validateConnState(ctx context.Context, cs *tls.ConnectionState) (returnerr
 		}
 	}
 
-	if len(leafCert.OCSPServer) == 0 {
-		errs = append(errs, fmt.Errorf("no OCSP server presented in leaf cert for %v", leafCert.Subject))
-		return
-	}
-
-	ocspResp, err := getOCSPResponse(ctx, leafCert.OCSPServer[0], leafCert, issuerCert)
-	if err != nil {
-		errs = append(errs, errors.Wrapf(err, "OCSP verification failed for %v", leafCert.Subject))
-		return
-	}
-
-	if ocspResp.Status == ocsp.Unknown {
-		errs = append(errs, fmt.Errorf("unknown OCSP verification status for %v", leafCert.Subject))
-	}
-
-	if ocspResp.Status == ocsp.Revoked {
-		errs = append(errs, fmt.Errorf("cert for %v has been revoked on %v, reason: %v", leafCert.Subject, ocspResp.RevokedAt, ocspResp.RevocationReason))
+	// Only check CRL if a distribution point is present in the leaf cert.
+	if len(leafCert.CRLDistributionPoints) > 0 {
+		revoked, err := checkCertCRL(ctx, leafCert.CRLDistributionPoints[0], leafCert, issuerCert)
+		if revoked && err != nil {
+			errs = append(errs, errors.Wrapf(err, "CRL verification failed for %v", leafCert.Subject))
+		}
 	}
 	return
 }
 
-func getOCSPResponse(ctx context.Context, ocspServer string, leafCert, issuerCert *x509.Certificate) (*ocsp.Response, error) {
-	reqb, err := ocsp.CreateRequest(leafCert, issuerCert, nil)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not create OCSP request")
-	}
-	hreq, err := http.NewRequestWithContext(ctx, "POST", ocspServer, bytes.NewReader(reqb))
+func checkCertCRL(ctx context.Context, crlURL string, leafCert, issuerCert *x509.Certificate) (bool, error) {
+	hreq, err := http.NewRequestWithContext(ctx, "GET", crlURL, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not create OCSP POST request")
+		return false, errors.Wrap(err, "could not create CRL GET request")
```

Comment:
Suggestion: Let’s switch over to [`%w` error wrapping](https://go.dev/blog/go1.13-errors#wrapping-errors-with-w) when we’re writing new code:

```suggestion
		return false, fmt.Errorf("could not create CRL GET request: %w", err)
```

No need to change any lines that you haven’t touched, though.